packages/x-date-pickers/src/internals/hooks/useField/useFieldState.ts

```diff
@@ -367,7 +354,7 @@ export const useFieldState = <
     const sections = fieldValueManager.getSectionsFromValue(
       utils,
       localeText,
-      state.sections,
+      null,
```

Comment:
We now can have an invalid date in state, and if the date is invalid and `prevSections` is defined, then we re-use the sections, which is not what we want here.